ads/google/a4a/performance.js

```diff
@@ -138,11 +56,47 @@ export class BaseLifecycleReporter {
    * @param {string} unusedName A descriptive name for the beacon signal.
    */
   sendPing(unusedName) {}
+
+  /**
+   * Set a URL parameter to be added to the ping data.  The parameter's value is
+   * subject to URL replacement and both parameter name and value are URI
+   * encoded before being written to the ping.  The entry is silently dropped
+   * if either `parameter` or `value` is falsey, with the exception that the
+   * `value` may be 0.
+   *
+   * @param {string} parameter
+   * @param {string|number} value
+   */
+  setPingParameter(parameter, value) {
+    if (parameter == null || parameter === '') {
+      return;
+    }
+    if (value === null || value === undefined || value === '') { return; }
+    this.extraVariables_[parameter] = String(value);
+  }
+
+  /**
+   * Sets a (possibly empty) collection of URL parameter values by invoking
+   * #setPingParameter on each key/value pair in the input collection.
+   *
+   * @param {!Object<string, string|number>} parametersToValues
+   */
+  setPingParameters(parametersToValues) {
+    for (const variable in parametersToValues) {
```

Comment:
FWIW Object.assign will almost do this for you except that you're using setPingParameter to clean the values...